File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/settings/di/SettingsComponent.java
package com.github.willjgriff.ethereumwallet.ui.settings.di;

import com.github.wiljgriff.ethereumwallet.di.AppComponent;
import com.github.wiljgriff.ethereumwallet.di.ControllerScope;
import com.github.willjgriff.ethereumwallet.ui.settings.ChangeAddressController;
import com.github.willjgriff.ethereumwallet.ui.settings.DeleteAddressAlertDialog;
import com.github.willjgriff.ethereumwallet.ui.settings.SettingsController;

import dagger.Component;

/**
 * Created by Will on 14/02/2017.
 */

@ControllerScope
@Component(dependencies = AppComponent.class)
public interface SettingsComponent {

	void inject(SettingsController settingsController);

	void inject(DeleteAddressAlertDialog alertDialog);

	void inject(ChangeAddressController changeAddressController);
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/settings/di/SettingsComponent.kt
package com.github.willjgriff.ethereumwallet.ui.settings.di

import com.github.willjgriff.ethereumwallet.di.AppComponent
import com.github.willjgriff.ethereumwallet.di.AppInjector
import com.github.willjgriff.ethereumwallet.di.ControllerScope
import com.github.willjgriff.ethereumwallet.ui.settings.ChangeAddressController
import com.github.willjgriff.ethereumwallet.ui.settings.DeleteAddressAlertDialog
import com.github.willjgriff.ethereumwallet.ui.settings.SettingsController
import dagger.Component

/**
 * Created by williamgriffiths on 04/05/2017.
 */
@ControllerScope
@Component(dependencies = arrayOf(AppComponent::class))
interface SettingsComponent {

    fun inject(settingsController: SettingsController)

    fun inject(alertDialog: DeleteAddressAlertDialog)

    fun inject(changeAddressController: ChangeAddressController)
}

fun SettingsController.injectPresenter() {
    getComponent().inject(this)
}

fun DeleteAddressAlertDialog.injectPresenter() {
    getComponent().inject(this)
}

fun ChangeAddressController.injectPresenter() {
    getComponent().inject(this)
}

private fun getComponent(): SettingsComponent {
    return DaggerSettingsComponent.builder()
            .appComponent(AppInjector.appComponent)
            .build()
}